src/base/static/components/utils/loader-with-spinner.js

```diff
@@ -0,0 +1,53 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+
+import Spinner from "react-spinner";
+
+import { translate } from "react-i18next";
+
+import "./loader-with-spinner.scss";
+
+class LoaderWithSpinner extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isLoaded: false,
+      isWithError: false,
+    };
+  }
+
+  componentWillMount() {
```

Comment:
This seems to come up in every code review - use `componentDidMount` instead.

Please read these:
https://reactjs.org/docs/react-component.html#componentdidmount
https://stackoverflow.com/a/27139507/1884158

Reply:
Gotcha-- I'll keep this in mind going forward.